src/transformers/models/grounding_dino/processing_grounding_dino.py

```diff
@@ -60,6 +65,57 @@ def get_phrases_from_posmap(posmaps, input_ids):
     return token_ids
 
 
+def _is_merged_candidate_labels_text(text: str) -> bool:
+    """
+    Check if the text is a merged candidate labels text. The general rule for constructing
+    candidate labels string is to separate the text labels with a dot. For example, "cat. dog."
+    """
+    if isinstance(text, str) and "." in text:
+        return True
+    return False
+
+
+def _is_list_of_candidate_labels(text) -> bool:
+    """Check that text is list/tuple of strings and each string is a candidate label and not merged candidate labels text."""
+    if isinstance(text, (list, tuple)):
+        return all(isinstance(t, str) and not _is_merged_candidate_labels_text(t) for t in text)
+    return False
+
+
+def _is_batched_list_of_candidate_labels(text) -> bool:
+    """Check that text is batch of list/tuple of strings and each string is a candidate label and not merged candidate labels text."""
+    if isinstance(text, (list, tuple)):
+        return all(_is_list_of_candidate_labels(t) for t in text)
+    return False
```

Comment:
same here why not just have 1 function with good comments?

Reply:
Removed `_is_batched_list_of_candidate_labels`